examples/relational_algebra.dx

```diff
@@ -0,0 +1,122 @@
+'# The Relational Algebra
+Unlike other languages which have bespoke "table"  or "dataframe" datatypes, in principle
+Dex's record system natively allows the manipulation of tables of records
+to do the sorts of data-munging usually done by packages such as pandas or dplyr.
+
+'This is an attempt to express the basic operations on datasets
+described by the relational algebra,
+roughly following [Wikipedia](https://en.wikipedia.org/wiki/Relational_algebra).
+
+'Right now there are two major differences from the standard relational algebra:
+1. We're not using sets of records, but rather tables.  I'm not sure which is better.
```

Comment:
This is fine. SQL also doesn't have set semantics, because every table entry has a distinct primary key (which corresponds to table's index set in Dex!).